xonsh/environ.py

```diff
@@ -120,6 +121,7 @@ def to_debug(x):
     'XONSH_ENCODING_ERRORS': (is_string, ensure_string, ensure_string),
     'XONSH_HISTORY_SIZE': (is_history_tuple, to_history_tuple, history_tuple_to_str),
     'XONSH_LOGIN': (is_bool, to_bool, bool_to_str),
+    'XONSH_SHOW_TRACEBACK': (is_tb_opt, to_tb_tuple, tb_tuple_to_str),
```

Comment:
That was my main point, yes.  I would be surprised if I set `$XONSH_SHOW_TRACEBACK = True` and then I checked its value later and it was `(True, None)` or similar.

I was not suggesting removing it entirely from the ensurers, but rather modifying how the detyping and retyping works.  If we're worried about the error reporting, we can have the ensurer print an error message if it is given an invalid value.